compiler/rustc_ast/src/mut_visit.rs

```diff
@@ -489,6 +489,9 @@ pub fn noop_visit_ty<T: MutVisitor>(ty: &mut P<Ty>, vis: &mut T) {
             visit_vec(bounds, |bound| vis.visit_param_bound(bound));
         }
         TyKind::MacCall(mac) => vis.visit_mac_call(mac),
+        TyKind::AnonymousStruct(fields, ..) | TyKind::AnonymousUnion(fields, ..) => {
```

Comment:
Please explicitly ignore fields in visitors, so we have a compiler error when we change them.
```suggestion
        TyKind::AnonymousStruct(fields, _recovered) | TyKind::AnonymousUnion(fields, _recovered) => {
```